### PR TITLE
#17914 Fix excess quoting when renaming target entity/attribute name

### LINKED ISSUE #17914: Some symbols( ` , ") in columns name are repeated twice during data import to new table
### Description

Some symbols( ` , ") in columns name are repeated twice during data import to a new table if the column name was changed during the import.

was founded during the check of #17619 

### DBeaver Version

22.2.2

### Operating System

win10

### Database and driver

mysql 5.7

### Steps to reproduce

1. Start import from csv to new table
2. Change some column name by adding, e.g. grave accent - We can see that grave accent is doubled in the mapping dialog
dbeaver_T1KRfjwi11

3. Check table DDL - there are four grave accent in DDL
imageimporting

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

**Comment by LonwoLonwo** [comment]
And it still not working for this case #18148

2022-11-22 12_36_43-Configure metadata structure

[one_column_with_header.csv](https://github.com/dbeaver/dbeaver/files/10066043/one_column_with_header.csv)

**Comment by ShadelessFox** [comment]
Turns out, we don't need to ever escape identifiers in the data transfer wizard. They will be escaped by the processor.

**Comment by E1izabeth** [comment]
Maybe we want to write unit tests also?

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java [hunk 1]
```diff
@@ -206,13 +206,24 @@ public boolean mustBeQuoted(String str, boolean forceCaseSensitive) {
...
     @Override
     public String escapeString(String string) {
-        return string.replace("'", "''").replace("`", "``").replaceAll("\\\\(?![_%?])", "\\\\\\\\");
+        return escapeString(string, null);
+    }
+
+    @NotNull
+    protected String escapeString(@NotNull String string, @Nullable String[] quotes) {
+        if (quotes != null) {
+            string = string.replace(quotes[0], quotes[0] + quotes[0]);
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:221]: I'm not sure about the idea behind this code, but maybe you wanted something like this?
```suggestion
            string = string.replace(quotes[0], quotes[0] + quotes[1]);
```

**Comment by ShadelessFox** [comment]
All changes were made within UI bundles. In any case, I'm not sure how to test it.

**Comment by LonwoLonwo** [comment]
> All changes were made within UI bundles.

This is not true.

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java [hunk 2]
```diff
@@ -206,13 +206,24 @@ public boolean mustBeQuoted(String str, boolean forceCaseSensitive) {
...
+
     @NotNull
     @Override
     public String escapeString(String string) {
-        return string.replace("'", "''").replace("`", "``").replaceAll("\\\\(?![_%?])", "\\\\\\\\");
+        return escapeString(string, null);
+    }
+
+    @NotNull
+    protected String escapeString(@NotNull String string, @Nullable String[] quotes) {
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:219]: You can at least create a dialect test if you do not know how to make a datatransfer test. We already have PostgreDialectTest and OracleDialectTest. Please look at them.

**Review by LonwoLonwo — CHANGES_REQUESTED** [review]

**Comment by serge-rider** [comment]
We need unit tests for that

File: test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MySQLDialectTest {
+    @Test
+    public void quoteStringTest() {
+        final MySQLDialect dialect = new MySQLDialect();
+
+        Assert.assertEquals("`a ' b`", dialect.getQuotedIdentifier("a ' b", false, true));
+        Assert.assertEquals("`a `` b`", dialect.getQuotedIdentifier("a ` b", false, true));
```
Comment (E1izabeth) [inline test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java:31]:
```suggestion
        Assert.assertEquals("`a ' b`", dialect.getQuotedIdentifier("a ' b", false, true));
        Assert.assertEquals("`a `` b`", dialect.getQuotedIdentifier("a ` b", false, true));
        Assert.assertEquals("a b", dialect.getQuotedIdentifier("a b", false, true));
```

Comment (ShadelessFox) [reply test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java:31]: 👍🏻

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java (hunk 2 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialect.java:219]: Done

**Review by E1izabeth — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**